### PR TITLE
Filter fixes

### PR DESCRIPTION
**Pull request by tristanhcole**
* frontend filter fixes
* filter now associated with organisation
* backend filter tests

### CODE REVIEW

**Comment by tristanhcole** [comment]
[Filters aren't working on the dashboard page](https://trello.com/c/Q40Ny85M/220-filters-arent-working-on-the-dashboard-page)

**Comment by tristanhcole** [comment]
This branch seems to break clear_dev_data due to filter being associated to organization and `drop_all` not cascading...

```
sqlalchemy.exc.InternalError: (psycopg2.InternalError) cannot drop table organisation because other objects depend on it
DETAIL:  constraint saved_filter_organisation_id_fkey on table saved_filter depends on table organisation
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
```

File: app/server/api/filter_api.py [hunk 1]
```diff
@@ -51,11 +51,12 @@ def post(self):
...
         create_filter = SavedFilter(
             name=filter_name,
-            filter=filter_attributes
+            filter=filter_attributes,
+            organisation_id=g.user.default_organisation_id
         )
 
         db.session.add(create_filter)
-        db.session.commit()
+        db.session.flush()
```
Comment (enjeyw) [inline app/server/api/filter_api.py]: At this point, i feel like there's not a lot of reason to have the commit in the after_request hook :|

Comment (tristanhcole) [reply app/server/api/filter_api.py]: i still like it

**Comment by tristanhcole** [comment]
> This branch seems to break clear_dev_data due to filter being associated to organization and `drop_all` not cascading...
> 
> ```
> sqlalchemy.exc.InternalError: (psycopg2.InternalError) cannot drop table organisation because other objects depend on it
> DETAIL:  constraint saved_filter_organisation_id_fkey on table saved_filter depends on table organisation
> HINT:  Use DROP ... CASCADE to drop the dependent objects too.
> ```

do we still use `clear_dev_data` @enjeyw ?

**Pull request merged**